### PR TITLE
Update JSON functions

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Add a clarifying sentence and update the title for the JSON functions section and add missing entries into the list pages.

## Additional context and related issues

* https://trinodb.slack.com/archives/CP1MUNEUX/p1674102698927119

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: docs/src/main/sphinx/functions/list-by-topic.rst [hunk 1]
```diff
@@ -316,11 +316,15 @@ For more details, see :doc:`json`
 
 * :func:`is_json_scalar`
 * :func:`json_array_contains`
+* :ref:`json_array() <json_array>`
+* :func:`json_array_get`
```
Comment (ebyhr) [inline docs/src/main/sphinx/functions/list-by-topic.rst:320]: What's the order policy here? I thought `json_array` should be before `json_array_contains`.

**Review by ebyhr — APPROVED** [review]

**Comment by findepi** [comment]
docs build failed

**Comment by mosabua** [comment]
> docs build failed

Whosa .. weird.. let me check

File: docs/src/main/sphinx/functions/list-by-topic.rst (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/functions/list-by-topic.rst]: Oops .. fixed

**Comment by mosabua** [comment]
I updated and verified that it builds locally. Should be good to go @findepi and @ebyhr

**Comment by mosabua** [comment]
And CI now also passed

**Comment by mosabua** [comment]
Thanks @ebyhr !

**Pull request merged**